google/resource_google_service_account.go

```diff
@@ -33,9 +34,10 @@ func resourceGoogleServiceAccount() *schema.Resource {
 				Computed: true,
 			},
 			"account_id": &schema.Schema{
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(6, 30),
```

Comment:
Since we do have some more information on what a valid id is, let's just go ahead and check the regexp: https://cloud.google.com/iam/reference/rest/v1/projects.serviceAccounts/create
> It is unique within a project, must be 6-30 characters long, and match the regular expression [a-z]([-a-z0-9]*[a-z0-9]) to comply with RFC1035.

Reply:
Done